internal/domain/review_suggestion.go

```diff
@@ -0,0 +1,27 @@
+package domain
+
+import (
+	"github.com/invopop/jsonschema"
+)
+
+type ReviewSuggestion struct {
+	Comment  string `json:"comment"`
+	Position int    `json:"position"`
+}
+
+type ReviewSuggestionsResponse struct {
+	ReviewSuggestions []ReviewSuggestion `json:"review_suggestions"`
+}
+
+func GenerateSchema[T any]() interface{} {
+	reflector := jsonschema.Reflector{
+		AllowAdditionalProperties: false,
+		DoNotReference:            true,
+	}
+	var v T
+	schema := reflector.Reflect(v)
+	return schema
```

Comment:
Consider making the `ReviewSuggestionSchema` variable private if it's intended for internal use only. This can prevent unintended uses in other packages and improve encapsulation.